Make navbar slidable

nicholasreynolds opened the pull request:
A more ergonomic display of the navbar. There are still plenty of bugs to be worked out, but I've gotten most of the big ones.

With the large number of menu items in our navbar, this design is definitely needed. Like I said, it's not complete, but now we can work on it together. I have the sliding animations done and it is still fixed on scroll, but takes up less space.

![image](https://cloud.githubusercontent.com/assets/7495152/19616154/db023c00-97c0-11e6-8fbb-d01e18c3acec.png)

![image](https://cloud.githubusercontent.com/assets/7495152/19616157/e59bc1fe-97c0-11e6-91ef-3b37bb1b18c0.png)

![image](https://cloud.githubusercontent.com/assets/7495152/19616159/f03b1790-97c0-11e6-8669-d73fbd750c7a.png)

![image](https://cloud.githubusercontent.com/assets/7495152/19616166/16e229b0-97c1-11e6-8d8d-8fb1dc4823ec.png)

nicholasreynolds committed:
Make navbar slidable

dglin commented:
I like it a lot! I am sure you are working on these, but from what I can see:
-  Some kind of arrow in order to show users to click on the menu
  -   menu needs to extend all the way across the page
  -   also do you think sliding left -> right is better than sliding top -> bottom?

nicholasreynolds commented:
Sliding top to bottom is something I experimented with at first, for a
while, but It isn't practical and actually more cpu intensive   right to
left, since with it right->left we don't need to create a placeholder
element for the navbar. This, in effect, preemptively negates any future
bugs with jumping content. On my local machine I have this working, but
there is still some jumping content on my latest commit. This
experimentation is why it's taken me so long to commit.

On Mon, Oct 24, 2016, 11:54 AM David Lin notifications@github.com wrote:

> I like it a lot! I am sure you are working on these, but from what I can
> see:
> - Some kind of arrow in order to show users to click on the menu
>   - menu needs to extend all the way across the page
>   - also do you think sliding left -> right is better than sliding
>     top -> bottom?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/dglin/SJ-Theming/pull/11#issuecomment-255831703, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AHJd8IDConMQxv35nq-1tqlYx4903Ywuks5q3P7VgaJpZM4Kdul1
> .

saileshpatnala commented:
I like the idea and the sliding from left to right. 
- My only concern is that this design could potentially be counter-intuitive for some people but let's see what Prof. Joseph says.
- What do you guys think about having the navbar be fully present on landing and then condensing into the small menu button that Nicholas made when scrolling? Would that be too much?

nicholasreynolds committed:
Stylize navbar
	- Animate placholder height to zero (*Fixes* jumping content bug)
	- Change "Menu" text of trigger button to be chevrons instead

nicholasreynolds committed:
Make navbar return to initial state
